app/Domain/Integrations/Repositories/IntegrationUrlRepository.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Integrations\Repositories;
+
+use App\Domain\Integrations\IntegrationUrl;
+use Ramsey\Uuid\UuidInterface;
+
+interface IntegrationUrlRepository
+{
+    public function save(IntegrationUrl $integrationUrl): void;
+    public function update(IntegrationUrl $integrationUrl): void;
+    /**
+    * @param array<IntegrationUrl> $integrationUrls
+     */
+    public function updateUrls(array $integrationUrls): void;
+    public function getById(UuidInterface $id): IntegrationUrl;
+    /**
+    * @param array<UuidInterface> $ids
+    * @return array<IntegrationUrl>
+     */
+    public function getByIds(array $ids): array;
```

Comment:
I noticed we have a mix of `PaginatedCollection` and `array`. To be consistent I would replace the `array` with `Collection`.

Maybe something for a follow-up pull request?

Reply:
Let's do it in a follow-up PR